### PR TITLE
Add Fhir authentication to protect EHR writeback

### PR DESCRIPTION
**Pull request by flowy11**

### CODE REVIEW

File: app.js
Comment (dzelemba) [inline app.js]: 401 actually represents "unauthenticated" - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401 which isn't appropriate here because it's an authorization issue not authentication issue.

403 or 400 would be better.

**Review by dzelemba — COMMENTED** [review]
Looks good, just one small nit.

File: app.js
Comment (flowy11) [reply app.js]: Done.

Comment (dzelemba) [inline app.js]: This is still returning 401.

File: app.js
Comment (Biao) [inline app.js:103]: why not use Authorization header to pass the fhirAccessToken?

Comment (flowy11) [reply app.js:103]: The access token should be paired with a fhir base URL. I think it's better to put them at the same place, i.e., HTTP body.

File: app.js
Comment (flowy11) [reply app.js]: Done.

**Review by dzelemba — APPROVED** [review]
LGTM. Will wait for Biao to approve before merging.

File: app.js
Comment (Biao) [reply app.js:103]: I see. It doesn't feel like a common practice though. Anyway, it is fine. I was just curious.

**Review by Biao — APPROVED** [review]

**Pull request merged**
